### ISSUE TITLE
Infinite loop in getElementAtIndex

### ISSUE THREAD

**Issue by gaearon** [bvaughn/react-devtools-experimental#179]
This sequence of steps produces an infinite loop in `getElementAtIndex`.

```js
    ReactDOM.render(<Root>{null}</Root>, container)
    setTimeout(() => {
      ReactDOM.render(<div />, container)
    }, 1000)
```

**Comment by gaearon** [comment]
I'll probably assign you because I'm not sure what the right fix is. But in addition to the fix, there should probably be something to break out of the loop, at least as an assertion.

**Comment by gaearon** [comment]
The loop happens here:

https://github.com/bvaughn/react-devtools-experimental/blob/456e937b83b8c05b0f7cdd6ea0e464018f5d9f19/src/devtools/store.js#L320

The last element is just `<div>` so a root exists but doesn't show up (as it has no composite children).

**Comment by bvaughn** [comment]
Nice catch!

Fixed via d6e829d
